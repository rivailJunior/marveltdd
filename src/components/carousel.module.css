.showCarousel {
    display: block;
}

.hideCarousel {
    display: none;
}