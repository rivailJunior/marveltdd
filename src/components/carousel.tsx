import { useState } from "react";
import { Carousel } from "react-bootstrap";
import styles from "./carousel.module.css";

type iCarouselProps = {
    data: [any];
    type: string;
};
export function CarouselInfor({ data, type }: iCarouselProps): JSX.Element {
    const [index, setIndex] = useState(0);

    const handleSelect = (selectedIndex: number, e: any) => {
        setIndex(selectedIndex);
    };

    const showErr = !data.length ? styles.showCarousel : styles.hideCarousel;
    const showCarousel = data.length
        ? styles.showCarousel
        : styles.hideCarousel;
    return (
        <>
            <div className={showErr}>Ops. Nenhuma {type} encontrado</div>
            <Carousel
                className={showCarousel}
                activeIndex={index}
                onSelect={handleSelect}
            >
                {data.map((item, index) => {
                    <Carousel.Item key={index}>
                        <img
                            className="d-block w-100"
                            src="holder.js/800x400?text=First slide&bg=373940"
                            alt="First slide"
                        />
                        <Carousel.Caption>
                            <h3>First slide label</h3>
                            <p>
                                Nulla vitae elit libero, a pharetra augue mollis
                                interdum.
                            </p>
                        </Carousel.Caption>
                    </Carousel.Item>;
                })}
            </Carousel>
        </>
    );
}
