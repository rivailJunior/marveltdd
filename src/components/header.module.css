.marvelHeader {
    /* padding: 10%; */
    top: 0px;
    left: 0px;
    width: 100%;
    height: 64px;
    background: #FFFFFF 0% 0% no-repeat padding-box;
    opacity: 1;
}

.labelName {
    text-align: left;
    font: normal normal bold 14px/20px PT Sans Caption;
    letter-spacing: 0px;
    color: #555555;
}

.informationProps {
    height: 100%;
    width: 100%;
    background-color: lightcoral;
}