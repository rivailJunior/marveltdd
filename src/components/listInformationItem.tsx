import styles from "./marvelList.module.css";

type iListProps = {
    list: [any] | any;
    tagName: string;
    information?: string;
};
export const ListInformationItem = ({
    list,
    tagName,
    information,
}: iListProps): JSX.Element => {
    return list?.length > 0 ? (
        <ul className={styles.list}>
            {list.slice(0, 3).map((item: any, index: number) => {
                return (
                    <li data-testid="informationLi" key={index}>
                        {item[tagName]}
                    </li>
                );
            })}
        </ul>
    ) : (
        <div className={styles.defaultLabel}>
            Ops. {information} n√£o encontrados!
        </div>
    );
};
