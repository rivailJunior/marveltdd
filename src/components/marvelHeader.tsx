import { Navbar } from "react-bootstrap";
import styles from "./header.module.css";

type iHeaderProps = {
    leftLabel?: string;
    userName?: string;
    descriptionRight?: string;
};

const MarvelHeader = ({
    leftLabel,
    userName,
    descriptionRight,
}: iHeaderProps): JSX.Element => {
    const NoPropsDivInfor = () => {
        return (
            <div
                className={styles.informationProps}
                data-testid="informationProps"
            >
                ops!
            </div>
        );
    };

    return (
        <Navbar
            expand="lg"
            variant="light"
            bg="light"
            className={styles.marvelHeader}
        >
            <Navbar.Brand href="#home">
                {!leftLabel?.length ? NoPropsDivInfor() : leftLabel}
            </Navbar.Brand>
            <Navbar.Toggle />

            <Navbar.Collapse className="justify-content-end">
                <Navbar.Text className={styles.labelName}>
                    {!userName?.length ? NoPropsDivInfor() : userName}
                    {!descriptionRight?.length ? (
                        NoPropsDivInfor()
                    ) : (
                        <a href="#login">{descriptionRight}</a>
                    )}
                </Navbar.Text>
            </Navbar.Collapse>
        </Navbar>
    );
};

export { MarvelHeader };
