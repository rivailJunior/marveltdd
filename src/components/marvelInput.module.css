.showClear {
    display: block;
}

.hideClear {
    display: none !important;
}