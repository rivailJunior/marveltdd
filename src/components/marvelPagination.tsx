import { useState, useEffect } from "react";
import { Pagination } from "react-bootstrap";
import styles from "./marvelPagination.module.css";

type iPaginationProps = {
    total: number;
    handleActive?: Function;
};
const MarvelPagination = ({
    total,
    handleActive,
}: iPaginationProps): JSX.Element => {
    const [active, setactive] = useState(0);
    useEffect(() => {
        if (active > 0) {
            handleActive(active);
        }
    }, [active, handleActive]);
    const totalValues = Array(Math.ceil(total / 10)).fill(0);
    const isToHidePagination =
        total > 10 ? styles.showPagination : styles.noPagination;

    return (
        <Pagination className={isToHidePagination} data-testid="paginationDiv">
            <Pagination.First onClick={() => setactive(0)} />
            <Pagination.Prev onClick={() => setactive(active - 1)} />
            {totalValues.map((item, index) => {
                return (
                    index < 10 && (
                        <Pagination.Item
                            data-testid="paginationItems"
                            key={index}
                            active={index === active}
                            onClick={() => setactive(index + 1)}
                        >
                            {index + 1}
                        </Pagination.Item>
                    )
                );
            })}
            <Pagination.Next onClick={() => setactive(active + 1)} />
            <Pagination.Last onClick={() => setactive(totalValues.length)} />
        </Pagination>
    );
};

export { MarvelPagination };
