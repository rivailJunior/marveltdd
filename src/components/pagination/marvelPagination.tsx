import { useState } from "react";
import { Pagination } from "react-bootstrap";
import styles from "./marvelPagination.module.css";

type iPaginationProps = {
    total: number;
    handleActive?: Function;
    currentPage: number
};
const MarvelPagination = ({
    total,
    handleActive,
    currentPage
}: iPaginationProps): JSX.Element => {
    const [active, setActive] = useState(currentPage);
    const [maxAndMin, setMaxAndMin] = useState({ max: 9, min: 0 })
    const totalValues = Array(Math.ceil(total / 10)).fill(0);
    const isToHidePagination =
        total > 10 ? styles.showPagination : styles.noPagination;

    const setPaginationOffset = (current: number) => {
        if (current >= 9) {
            setMaxAndMin({ max: current + 1, min: current - 9 })
        } else {
            setMaxAndMin({ max: 9, min: 0 })
        }
    }

    return (
        <Pagination className={isToHidePagination} data-testid="paginationDiv">
            {active > 0 && active > 1 && (
                <Pagination.First
                    className={styles.paginationItem}
                    onClick={() => {
                        setActive(0);
                        handleActive(0);
                        setPaginationOffset(0)
                    }}
                />
            )}
            {active > 0 && (
                <Pagination.Prev
                    className={styles.paginationItem}

                    onClick={() => {
                        const current = active === 1 ? 0 : active - 1;
                        setActive(current);
                        handleActive(current);
                        setPaginationOffset(current)
                    }}
                />
            )}

            {totalValues.map((item, index) => {
                return index >= maxAndMin.min && index <= maxAndMin.max && (
                    <Pagination.Item
                        data-testid="paginationItems"
                        key={index}
                        active={index === active}
                        className={index === active ? ` ${ styles.paginationItem } ${ styles.active } ${ styles.paginate } paginationActive` : `${ styles.paginationItem } ${ styles.paginate }`}
                        onClick={() => {
                            setActive(index);
                            handleActive(index);
                            setPaginationOffset(index)
                        }}
                    >
                        {index + 1}
                    </Pagination.Item>

                );
            })}
            {active !== totalValues.length - 1 && (
                <Pagination.Next
                    className={styles.paginationItem}
                    onClick={() => {
                        const current =
                            active === totalValues.length ? active : active + 1;
                        setActive(current);
                        handleActive(current);
                        setPaginationOffset(current)
                    }}
                />
            )}

            {active > 0 && active !== totalValues.length - 1 && (
                <Pagination.Last
                    className={styles.paginationItem}
                    onClick={() => {
                        setActive(totalValues.length - 1);
                        handleActive(totalValues.length - 1);
                        setPaginationOffset(totalValues.length - 1)
                    }}
                />
            )}
        </Pagination>
    );
};

export { MarvelPagination };
