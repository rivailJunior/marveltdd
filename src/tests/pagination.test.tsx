import React from "react";
import { render, cleanup, fireEvent } from "@testing-library/react";
import renderer from "react-test-renderer";
import { MarvelPagination } from "../components/marvelPagination";

afterEach(cleanup);

describe("`<MarvelPagination>`", () => {
    test("Render correctly", () => {
        const container = renderer.create(<MarvelPagination total={100} />);
        container.toJSON();
        expect(container).toMatchSnapshot();
    });

    test("Hide pagination if total is less than 10", () => {
        const { getByTestId } = render(<MarvelPagination total={9} />);
        const pagination = getByTestId("paginationDiv");
        expect(pagination).toHaveClass("noPagination");
    });

    test("Show max 10 pagination itens if list is too higher", () => {
        const { getAllByTestId } = render(<MarvelPagination total={1000} />);
        const pagination = getAllByTestId("paginationItems");
        expect(pagination).toHaveLength(10);
    });

    test("Render with first item as active", () => {
        const { getAllByTestId } = render(
            <MarvelPagination total={1000} handleActive={() => {}} />
        );
        const paginationItem = getAllByTestId("paginationItems");
        expect(paginationItem[0].parentNode).toHaveClass("active");
    });

    test("Set new item as active", () => {
        const callback = jest.fn();
        const { getByText } = render(
            <MarvelPagination total={1000} handleActive={callback} />
        );
        fireEvent.click(getByText(/5/i));
        expect(callback).toBeCalledTimes(1);
        expect(callback).toBeCalledWith(5);
    });
});
