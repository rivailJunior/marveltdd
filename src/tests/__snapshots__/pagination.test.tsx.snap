// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`<MarvelPagination>\` Render correctly 1`] = `
<ul
  className="showPagination pagination"
  data-testid="paginationDiv"
>
  <li
    className="page-item"
  >
    <a
      className="page-link"
      href="#"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
    >
      <span
        aria-hidden="true"
      >
        «
      </span>
      <span
        className="sr-only"
      >
        First
      </span>
    </a>
  </li>
  <li
    className="page-item"
  >
    <a
      className="page-link"
      href="#"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
    >
      <span
        aria-hidden="true"
      >
        ‹
      </span>
      <span
        className="sr-only"
      >
        Previous
      </span>
    </a>
  </li>
  <li
    className="page-item active"
  >
    <span
      className="page-link"
      data-testid="paginationItems"
      disabled={false}
      onClick={[Function]}
    >
      1
      <span
        className="sr-only"
      >
        (current)
      </span>
    </span>
  </li>
  <li
    className="page-item"
  >
    <a
      className="page-link"
      data-testid="paginationItems"
      href="#"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
    >
      2
    </a>
  </li>
  <li
    className="page-item"
  >
    <a
      className="page-link"
      data-testid="paginationItems"
      href="#"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
    >
      3
    </a>
  </li>
  <li
    className="page-item"
  >
    <a
      className="page-link"
      data-testid="paginationItems"
      href="#"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
    >
      4
    </a>
  </li>
  <li
    className="page-item"
  >
    <a
      className="page-link"
      data-testid="paginationItems"
      href="#"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
    >
      5
    </a>
  </li>
  <li
    className="page-item"
  >
    <a
      className="page-link"
      data-testid="paginationItems"
      href="#"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
    >
      6
    </a>
  </li>
  <li
    className="page-item"
  >
    <a
      className="page-link"
      data-testid="paginationItems"
      href="#"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
    >
      7
    </a>
  </li>
  <li
    className="page-item"
  >
    <a
      className="page-link"
      data-testid="paginationItems"
      href="#"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
    >
      8
    </a>
  </li>
  <li
    className="page-item"
  >
    <a
      className="page-link"
      data-testid="paginationItems"
      href="#"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
    >
      9
    </a>
  </li>
  <li
    className="page-item"
  >
    <a
      className="page-link"
      data-testid="paginationItems"
      href="#"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
    >
      10
    </a>
  </li>
  <li
    className="page-item"
  >
    <a
      className="page-link"
      href="#"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
    >
      <span
        aria-hidden="true"
      >
        ›
      </span>
      <span
        className="sr-only"
      >
        Next
      </span>
    </a>
  </li>
  <li
    className="page-item"
  >
    <a
      className="page-link"
      href="#"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
    >
      <span
        aria-hidden="true"
      >
        »
      </span>
      <span
        className="sr-only"
      >
        Last
      </span>
    </a>
  </li>
</ul>
`;
